spring-ai-model/src/main/java/org/springframework/ai/model/tool/DefaultToolCallingChatOptions.java

```diff
@@ -46,6 +47,9 @@ public class DefaultToolCallingChatOptions implements ToolCallingChatOptions {
 	@Nullable
 	private Boolean internalToolExecutionEnabled;
 
+	@Nullable
+	private Integer internalToolExecutionMaxAttempts = ToolCallingChatOptions.DEFAULT_TOOL_EXECUTION_MAX_ATTEMPTS;
```

Comment:
Consider renaming this to something like `toolExecutionMaxIterations`. 

I'm suggesting dropping the `internal` prefix since this option can also be used as part of the _user-controller tool execution_, which is executed externally from the `ChatModel`.

Also, I'm suggesting replacing `attempts` with `iterations` since attempts would make seem related to the concept of retrying tool calls, whereas here we want to limit the number of iterations in the loop supporting the tool call requests from the model.